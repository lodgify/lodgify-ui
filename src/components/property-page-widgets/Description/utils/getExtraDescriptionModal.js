import React from 'react';
import { Button } from 'semantic-ui-react';

import { Modal } from 'elements/Modal';

/**
 * @param  {string|Object} extraDescription
 * @param  {string} extraDescriptionButtonText
 * @return {Object}
 */
export const getExtraDescriptionModal = (
  extraDescription,
  extraDescriptionButtonText
) => (
  <Modal
    hasPadding
    trigger={<Button basic>{extraDescriptionButtonText}</Button>}
  >
    {extraDescription}
  </Modal>
);
