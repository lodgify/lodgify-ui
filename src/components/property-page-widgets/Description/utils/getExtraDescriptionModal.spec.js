import { getExtraDescriptionModal } from './getExtraDescriptionModal';

describe('getExtraDescriptionModal', () => {
  it('should render the right structure', () => {
    const actual = getExtraDescriptionModal('button', 'text');

    expect(actual).toMatchSnapshot();
  });
});
