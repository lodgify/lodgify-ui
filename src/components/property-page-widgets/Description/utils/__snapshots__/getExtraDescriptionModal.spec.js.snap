// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getExtraDescriptionModal should render the right structure 1`] = `
<Modal
  hasCloseIcon={true}
  hasPadding={true}
  hasRoundedCorners={false}
  header={null}
  isFullscreen={false}
  onClose={[Function]}
  size="tiny"
  trigger={
    <Button
      as="button"
      basic={true}
    >
      text
    </Button>
  }
>
  button
</Modal>
`;
